### PR TITLE
Various output channel-related fixes

### LINKED ISSUE #8122: [output][vscode][plugin] Output channels are not cleaned up after closing the channel
### Bug Description:

The output channel registry for the extensions (VS Code and Theia plug-in) does its own caching:
https://github.com/eclipse-theia/theia/blob/250841db7f5de49a1c57ffb437379c8638d35be0/packages/plugin-ext/src/main/browser/output-channel-registry-main.ts#L93

Disposing the channel on the FE does not update the channel cache for the extensions, and tries to work on a disposed editor model.

```
textModel.ts:360 Uncaught (in promise) Error: Model is disposed!
    at TextModel._assertNotDisposed (textModel.ts:360)
    at TextModel.getLineCount (textModel.ts:739)
    at OutputChannel.<anonymous> (output-channel.ts:422)
    at step (output-channel.ts:15)
    at Object.next (output-channel.ts:15)
    at fulfilled (output-channel.ts:15)
```

### Steps to Reproduce:

1. Start a channel from an extension: `const c = (theia|vscode).window.createOutputChannel('test-channel'); c.appendLine('running...');`
2. Close the channel with the _Output: Close Output Channel..._ command.
3. Repeat step 1.

### Additional Information

- Operating System:
- Theia Version:

### PR DESCRIPTION
**Pull request by kittaakos**
#### What it does

 - Clear the text modification queue when disposing the output channel,
 - Exposed `private` `OutputChannel` properties with `protected` visibility.
 - From now on, downstreams can customize the output channel instance creation with a template method.
 - Removed the superfluous channel caching for VS Code extensions. Instead of erroneously keeping an extra cache of output channels for the extensions, we delegate into Theia's singleton channel manager via the command API. (Closes #8122)

#### How to test

 - This is where it gets tricky, I am way behind with the Theia versions. I still use `@theia/languages`.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by kittaakos** [comment]
@akosyakov, please take a look and provide some upfront feedback. Thanks!

**Comment by kittaakos** [comment]
@benoitf, as the original committer of the [`output-channel-registry-main.ts`](https://github.com/eclipse-theia/theia/pull/2007/files#diff-11ebe3cff728b3126a471050a7378864) module, please review.

**Review by benoitf — APPROVED** [review]
sorry @kittaakos I'm not the original author https://github.com/theia-demo-plugins/wiptheia/blame/b9a2ee833b201ae0c07f4245ab75060be02f1d6c/packages/plugin-ext/src/main/browser/output-channel-registry-main.ts

but plugin's output is there 
image

**Comment by kittaakos** [comment]
> sorry @kittaakos I'm not the original author https://github.com/theia-demo-plugins/wiptheia/blame/b9a2ee833b201ae0c07f4245ab75060be02f1d6c/packages/plugin-ext/src/main/browser/output-channel-registry-main.ts

Uhh, sorry for the confusion, @benoitf. GH has tricked me.

> @benoitf, as the original committer

output

Thank you for looking into it. 👍

**Comment by kittaakos** [comment]
I am going to leave it open until next week, then merge it if there is no objection.

**Pull request merged**
